archetype-common/src/main/java/org/apache/maven/archetype/common/util/PomUtils.java

```diff
@@ -79,6 +83,13 @@ public static boolean addNewModule( String artifactId, Reader fileReader, Writer
         dbf.setXIncludeAware( false );
         dbf.setExpandEntityReferences( false );
 
+        InputStream inputStream = PomUtils.class.getClassLoader().getResourceAsStream( "maven-4.0.0.xsd" );
```

Comment:
What is the purpose of loading this schema?

Reply:
Please see my minimal, reproducible example here https://issues.apache.org/jira/browse/ARCHETYPE-584?focusedCommentId=17008264&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17008264

I investigated this 1.5 years ago and can not remember the details why it only worked this way. However, I am pretty sure that I did not dive into the implementation details of `DocumentBuilderFactory`, which is fed with the `schema`, so not sure if I would have been able to explain it very deep back then.

Reply:
My guess is that you don't need the schema to fix this bug. In fact, you probably only need `setIgnoringElementContentWhitespace` though `setNamespaceAware( true )` is always a good idea.

Reply:
Please check the [example](https://issues.apache.org/jira/browse/ARCHETYPE-584?focusedCommentId=17008264&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17008264) I provided. You will see that without the schema the output is not as expected.